solutions/usaco-576.mdx

```diff
@@ -0,0 +1,104 @@
+---
+id: usaco-576
+source: USACO Platinum 2015 December
+title: Max Flow
+author: Dustin Miao
+---
+
+**Time Complexity**: $\mathcal O(m\log n + n)$
+
+For each path $(a, b)$, we can split it into two paths: $(a, lca(a, b)), (b, lca(a, b))$. We 
+can find the LCA with any method, but the easiest is with binary lifting. For each node, 
+keep track of the number paths that begin at that node and the number that end at that node. 
+
+If $k$ is the number of paths coming from a node's children, $s$ is the number of paths 
+starting at the current node, and $e$ is the number of paths ending at the current node, 
+then the answer for that node is $k + s - e / 2$. The $e / 2$ is because by splitting up
+the paths, we double count the lca for each path. The number of paths going to the node's
+parent is $k + s - e$. Thus, we can process all the paths at once using a single post-order 
+traversal of the tree. 
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>
+#include <vector>
+using namespace std;
+
+int N, M;
+vector<int> adj[50005];
+
+int dep[50005], par[50005][20];
+
+void init_bl(int u, int p, int d) {
```

Comment:
one letter name here as well

Reply:
honestly, this seems fine to me. it's pretty clear what these mean if you look at the contents of the function

Reply:
looking at these, i have no clue what they do

Reply:
(optional) since this is plat, but if u want u can fix it.

Reply:
oops, should make it clear that `bl` means binary lifting.

Reply:
still don't know what u p and d mean